shell/platform/tizen/channels/key_event_channel.cc

```diff
@@ -261,15 +81,44 @@ void KeyEventChannel::SendKey(const char* key,
                               uint32_t scan_code,
                               bool is_down,
                               std::function<void(bool)> callback) {
-  auto iter1 = kSymbolToScanCode.find(key);
-  if (iter1 != kSymbolToScanCode.end()) {
-    scan_code = iter1->second;
+  uint64_t sequence_id = last_sequence_id_++;
+
+  PendingEvent pending;
+  // This event is sent through the embedder API (KeyEvent) and the platform
+  // channel (RawKeyEvent) simultaneously, and |callback| will be called once
+  // responses are received from both of them.
+  pending.unreplied = 2;
+  pending.any_handled = false;
+  pending.callback = std::move(callback);
+
+  if (pending_events_.size() > kMaxPendingEvents) {
```

Comment:
Does the process of sending the key continue even if an error condition is reached?

Reply:
Yes, only error messages are generated. I'm not sure if this check is really necessary but I referred to the Windows embedder code.